saleor/webhook/observability/obfuscation.py

```diff
@@ -47,6 +48,16 @@ def hide_sensitive_headers(
     }
 
 
+def obfuscate_url(url: str) -> str:
+    parts = urlparse(url)
+    if parts.username is None:
+        return url
```

Comment:
RFC 1738 says empty username is valid (it's not no user, but empty username)
```
>>> urlparse('https://:passwd@tst.com/foo').password
passwd
```

Reply:
@NyanKiyoshi that is a great point, but this case is handled. If the user in URL is empty 
`urlparse("https://:passwd@tst.com/foo").username` returns an empty string, not `None`. So if username is `None`, that means there are no credentials in the URL. It is checked in the test case: 
```
"https://:password@example.com/test" -> "https://:{MASK}@example.com/test"
```

Reply:
Indeed, I missed this. Maybe you could add a comment above the line to make it more explicit?

Reply:
done 😄